src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaSubmissionSelectionService.java

```diff
@@ -60,46 +63,52 @@ private record ResponseDTO(@JsonProperty("data") long submissionId // submission
         athenaRestTemplate.setRequestFactory(requestFactory);
         // Create connector
         connector = new AthenaConnector<>(log, athenaRestTemplate, ResponseDTO.class);
+        this.submissionRepository = submissionRepository;
     }
 
     /**
      * Fetches the proposedTextSubmission for a given exercise from Athena
      *
-     * @param exercise    the exercise to get the proposed Submission for
-     * @param submissions assessable submissions of the exercise
+     * @param exercise      the exercise to get the proposed Submission for
+     * @param submissionIds IDs of assessable submissions of the exercise
      * @return a Submission suggested by the Athena submission selector (e.g. chosen by the highest information gain)
      * @throws IllegalArgumentException if exercise isn't automatically assessable
      */
-    public Optional<TextSubmission> getProposedSubmission(TextExercise exercise, List<TextSubmission> submissions) {
+    public Optional<TextSubmission> getProposedSubmission(TextExercise exercise, List<Long> submissionIds) {
         if (!exercise.isFeedbackSuggestionsEnabled()) {
             throw new IllegalArgumentException("The Exercise does not have feedback suggestions enabled.");
         }
-        if (submissions.isEmpty()) {
+        if (submissionIds.isEmpty()) {
             return Optional.empty();
         }
 
         log.debug("Start Athena Submission Selection Service for Text Exercise '{}' (#{}).", exercise.getTitle(), exercise.getId());
 
-        log.info("Calling Remote Service to calculate next proposed submissions for {} submissions.", submissions.size());
+        log.info("Calling Remote Service to calculate next proposed submissions for {} submissions.", submissionIds.size());
 
         try {
-            final RequestDTO request = new RequestDTO(exercise, submissions.stream().mapToLong(TextSubmission::getId).boxed().toList());
+            final RequestDTO request = new RequestDTO(exercise, submissionIds);
             // TODO: make module selection dynamic (based on exercise)
             // allow no retries because this should be fast and it's not too bad if it fails
             ResponseDTO response = connector.invokeWithRetry(athenaUrl + "/modules/text/module_text_cofee/select_submission", request, 0);
             log.info("Remote Service to calculate next proposes submissions responded: {}", response.submissionId);
             if (response.submissionId == -1) {
                 return Optional.empty();
             }
-            return submissions.stream().filter(s -> s.getId() == response.submissionId).findFirst();
+            var submission = submissionRepository.findById(response.submissionId);
+            if (submission.isEmpty()) {
+                log.error("Athena returned a submission ID that does not exist in the database: {}", response.submissionId);
+                return Optional.empty();
+            }
+            return submission.map(s -> (TextSubmission) s);
```

Comment:
This returns the submission to later load the submission with results again. You can also just return the ID here and let the caller load the submission.

Reply:
Good catch, did that 👍